test/test_gradient_fwi.jl

```diff
@@ -61,4 +61,23 @@ dt = srcGeometry.dt[1]
 	@test isapprox(rate_1, 1.25f0; rtol=5f-2)
 	@test isapprox(rate_2, 1.5625f0; rtol=5f-2)
 
-end
\ No newline at end of file
+end
+
+
+@testset "fwi_objective correct algebra test with $(nlayer) layers and tti $(tti) and freesurface $(fs) and isic $(isic)" begin
```

Comment:
isic not tested as not part of the test suite setup, and the dft and checkpointing option are missing.